crates/uv-distribution/src/metadata/requires_dist.rs

```diff
@@ -97,48 +100,71 @@ impl RequiresDist {
+                            .map(|requirement| {
+                                match uv_pep508::Requirement::<VerbatimParsedUrl>::from_str(
+                                    requirement,
+                                ) {
+                                    Ok(requirement) => Ok(requirement),
+                                    Err(err) => Err(MetadataError::GroupParseError(
+                                        name.clone(),
+                                        requirement.clone(),
+                                        Box::new(err),
+                                    )),
+                                }
+                            })
+                            .collect::<Result<Vec<_>, _>>(),
+                    )
+                })
+                .chain(
+                    // Only add the `dev` group if `dev-dependencies` is defined
+                    dev_dependencies
+                        .into_iter()
+                        .map(|requirements| (DEV_DEPENDENCIES.clone(), Ok(requirements.clone()))),
+                )
+                .map(|(name, requirements)| {
+                    // Apply sources to the requirements
+                    match requirements {
+                        Ok(requirements) => match apply_source_strategy(
+                            source_strategy,
+                            requirements,
+                            &metadata,
                             project_sources,
                             project_indexes,
                             locations,
-                            project_workspace.workspace(),
+                            project_workspace,
                             lower_bound,
-                        )
-                        .map(move |requirement| match requirement {
-                            Ok(requirement) => Ok(requirement.into_inner()),
-                            Err(err) => {
-                                Err(MetadataError::LoweringError(requirement_name.clone(), err))
-                            }
-                        })
-                    })
-                    .collect::<Result<Vec<_>, _>>()?,
-                SourceStrategy::Disabled => dev_dependencies
-                    .into_iter()
-                    .map(uv_pypi_types::Requirement::from)
-                    .collect(),
-            };
-            if dev_dependencies.is_empty() {
-                BTreeMap::default()
-            } else {
-                BTreeMap::from([(DEV_DEPENDENCIES.clone(), dev_dependencies)])
-            }
+                            &name,
+                        ) {
+                            Ok(requirements) => Ok((name, requirements)),
+                            Err(err) => Err(err),
+                        },
+                        Err(err) => Err(err),
+                    }
+                })
+                .collect::<Result<Vec<_>, _>>()?;
+
+            dependency_groups.into_iter().collect::<BTreeMap<_, _>>()
```

Comment:
Does this correctly merge keys? Like if dev is included twice, does this concatenate the values? I sort of think it might replace them...

Reply:
I think it probably replaces them, yeah. I wasn't a goal to resolve that here (per the description) since this implemented before I wrote a proposal. I figured I'd write some test cases and address it separately for review purposes.